diff --git a/app/src/main/java/com/topjohnwu/magisk/ui/MainActivity.kt b/app/src/main/java/com/topjohnwu/magisk/ui/MainActivity.kt
index 0c7e64024..71fdb361e 100644
--- a/app/src/main/java/com/topjohnwu/magisk/ui/MainActivity.kt
+++ b/app/src/main/java/com/topjohnwu/magisk/ui/MainActivity.kt
@@ -168,7 +168,7 @@ class MainActivity : SplashActivity<ActivityMainMd2Binding>() {
         if (!Info.isEmulator && Info.env.isActive && System.getenv("PATH")
                 ?.split(':')
                 ?.filterNot { File("$it/magisk").exists() }
-                ?.any { File("$it/su").exists() } == true) {
+                ?.any { File("$it/suu").exists() } == true) {
             MagiskDialog(this).apply {
                 setTitle(R.string.unsupport_general_title)
                 setMessage(R.string.unsupport_other_su_msg)
diff --git a/native/jni/core/applets.cpp b/native/jni/core/applets.cpp
index 8d0d81446..41b8e8b01 100644
--- a/native/jni/core/applets.cpp
+++ b/native/jni/core/applets.cpp
@@ -10,7 +10,7 @@ using namespace std;
 
 using main_fun = int (*)(int, char *[]);
 
-constexpr const char *applets[] = { "su", "resetprop", "zygisk", nullptr };
+constexpr const char *applets[] = { "suu", "resetprop", "zygisk", nullptr };
 static main_fun applet_mains[] = { su_client_main, resetprop_main, zygisk_main, nullptr };
 
 static int call_applet(int argc, char *argv[]) {
diff --git a/native/jni/include/magisk.hpp b/native/jni/include/magisk.hpp
index 466e5f439..4ff87265a 100644
--- a/native/jni/include/magisk.hpp
+++ b/native/jni/include/magisk.hpp
@@ -28,7 +28,7 @@ extern std::string  MAGISKTMP;
 #define ZYGISKBIN   INTLROOT "/zygisk"
 #define SELINUXMOCK INTLROOT "/selinux"
 
-constexpr const char *applet_names[] = { "su", "resetprop", nullptr };
+constexpr const char *applet_names[] = { "suu", "resetprop", nullptr };
 
 #define POST_FS_DATA_WAIT_TIME       40
 #define POST_FS_DATA_SCRIPT_MAX_TIME 35
