diff --git a/core/src/main/java/com/topjohnwu/superuser/internal/BuilderImpl.java b/core/src/main/java/com/topjohnwu/superuser/internal/BuilderImpl.java
index ae18065..fff8a79 100644
--- a/core/src/main/java/com/topjohnwu/superuser/internal/BuilderImpl.java
+++ b/core/src/main/java/com/topjohnwu/superuser/internal/BuilderImpl.java
@@ -100,7 +100,7 @@ public final class BuilderImpl extends Shell.Builder {
         // Root mount master
         if (!hasFlags(FLAG_NON_ROOT_SHELL) && hasFlags(FLAG_MOUNT_MASTER)) {
             try {
-                shell = build("su", "--mount-master");
+                shell = build("suu", "--mount-master");
                 if (!shell.isRoot())
                     shell = null;
             } catch (NoShellException ignore) {}
@@ -109,7 +109,7 @@ public final class BuilderImpl extends Shell.Builder {
         // Normal root shell
         if (shell == null && !hasFlags(FLAG_NON_ROOT_SHELL)) {
             try {
-                shell = build("su");
+                shell = build("suu");
                 if (!shell.isRoot()) {
                     shell = null;
                 }
diff --git a/core/src/main/java/com/topjohnwu/superuser/internal/Utils.java b/core/src/main/java/com/topjohnwu/superuser/internal/Utils.java
index 1e3575f..7a5a0fb 100644
--- a/core/src/main/java/com/topjohnwu/superuser/internal/Utils.java
+++ b/core/src/main/java/com/topjohnwu/superuser/internal/Utils.java
@@ -151,7 +151,7 @@ public final class Utils {
             }
             // noinspection ConstantConditions
             for (String path : System.getenv("PATH").split(":")) {
-                File su = new File(path, "su");
+                File su = new File(path, "suu");
                 if (su.canExecute()) {
                     // We don't actually know whether the app has been granted root access.
                     // Do NOT set the value as a confirmed state.
